redhat-appstudio/service-provider-integration-operator
# feat: changing quay repository area to registry area

Signed-off-by: Pavol Baran <pbaran@redhat.com>

### What does this PR do?
Changes the valid permission areas for quay binding from `repository` and `repositoryMetadata` to `registry` and `registryMetadata` respectively.

Adds migration logic to the SPIAccessTokenBinding reconciler that changes the old areas into the new ones. 

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

![image](https://user-images.githubusercontent.com/73115616/203924380-616d3682-de2c-4a6d-957d-522351912919.png)


### How to test this PR?

1. Deploy with operator: `quay.io/pbaran/spio:SVPI-248-quay`
2. Create binding: 
```yaml
apiVersion: appstudio.redhat.com/v1beta1
kind: SPIAccessTokenBinding
metadata:
  name: test-binding-quay-without-protocol
  namespace: default
spec:
  permissions:
    required:
      - type: rw
        area: repository
      - type: rw
        area: repositoryMetadata
  repoUrl: quay.io/repository/yourorganization/image
  secret:
    type: kubernetes.io/basic-auth
```
3. Look at the created binding after a second or two and it should have changed permission areas to `registry` and `registryMetadata` and there should be no error with the binding.
4. Optionally, you can go through the oauth flow.
